### PR TITLE
Translucency support

### PR DESCRIPTION
**Pull request by trevorvanhoof**
You want to ignore resource.rc changes, and possibly you don't want the changes from 6ff7230.
Most risky change is the emissive & alpha reading code, only tested with Maya fbx exports at this point.

### CODE REVIEW

File: Shaders/basic_specgloss.fs [hunk 1]
```diff
@@ -124,5 +126,5 @@ void main(void)
 
   color += sample_colormap( map_emissive, out_texcoord ).rgb;
 
-  frag_color = vec4( pow( color * exposure, vec3(1. / 2.2) ), 1.0f );
+  frag_color = vec4( pow( color * exposure, vec3(1. / 2.2) ), alpha.w );
```
Comment (seece) [inline Shaders/basic_specgloss.fs:129]: Why .w? Seems like a typo
```suggestion
  frag_color = vec4( pow( color * exposure, vec3(1. / 2.2) ), alpha );
```

File: Shaders/pbr.fs [hunk 2]
```diff
@@ -182,10 +191,12 @@ vec3 sample_irradiance_fast( vec3 normal, vec3 vertex_tangent )
   // Sample the irradiance map if it exists, otherwise fall back to blurred reflection map.
   if ( has_tex_skyenv )
   {
-    vec2 polar = sphere_to_polar( normal );
+    vec2 polar = sphere_to_polar_clamp_y( normal, 180.0 );
     // HACK: Sample a smaller mip here to avoid high frequency color variations on detailed normal
     //       mapped areas.
-    float miplevel = 5.5; // tweaked for a 360x180 irradiance texture
+    // float miplevel = 5.5; // tweaked for a 360x180 irradiance texture
+	// ^ unhacked because the mip higher levels cause really weird shading around poles, try a diffuse sphere to see what I mean!
```
Comment (seece) [inline Shaders/pbr.fs:198]: No reason to leave the ugly hack comment there :D

File: Shaders/pbr.fs [hunk 3]
```diff
@@ -182,10 +191,12 @@ vec3 sample_irradiance_fast( vec3 normal, vec3 vertex_tangent )
...
+	float miplevel = 0.0;
     return textureLod( tex_skyenv, polar, miplevel ).rgb * exposure;
```
Comment (seece) [inline Shaders/pbr.fs:200]:
```suggestion
    return textureLod( tex_skyenv, polar, 0.0 ).rgb * exposure;
```

File: Shaders/pbr.fs [hunk 4]
```diff
@@ -398,7 +413,8 @@ void main(void)
     }
   }
 
-  vec3 color = ambient + Lo + emissive;
+  // Premultiplied alpha applied to the diffuse component only
+  vec3 color = (ambient + Lo) * alpha + emissive;
```
Comment (seece) [inline Shaders/pbr.fs:417]: "to the diffuse component only"? `ambient` and `Lo` both have also specular contribution so this comment seems misleading

File: src/Geometry.cpp [hunk 5]
```diff
@@ -492,32 +503,51 @@ void Geometry::Render( const glm::mat4x4 & _worldRootMatrix, Renderer::Shader *
   Renderer::SetShader( _shader );
 
   _shader->SetConstant( "global_ambient", mGlobalAmbient );
-  for ( std::map<int, Geometry::Node>::iterator it = mNodes.begin(); it != mNodes.end(); it++ )
+  
+  // TODO: Maybe we can cache the world matrices & 2 queues for opaque and transparant in LoadMesh
+  // but that depends if we want to add animation support (in which case we can't).
+  for(int j = 0; j < 2; ++j) // loop twice, first opaque, then transparent
```
Comment (seece) [inline src/Geometry.cpp:509]: I like this simple approach :)

**Review by seece — APPROVED** [review]
Thanks a lot for adding this! I added some nitpicks but it looks good.

File: Shaders/basic_specgloss.fs (hunk 1 above)
Comment (trevorvanhoof) [reply Shaders/basic_specgloss.fs]: Thanks, that is indeed a typo

File: Shaders/pbr.fs (hunk 4 above)
Comment (trevorvanhoof) [reply Shaders/pbr.fs]: Good find, I guess I was too hasty in added that, committed a version with only diffuse multiplication!

**Comment by Gargaj** [comment]
Could you also remove the .rc changes then too? I can't really merge selectively.

**Comment by trevorvanhoof** [comment]
There you go!

**Pull request merged**